src/core/ad-sizes.ts

```diff
@@ -371,6 +385,38 @@ const slotSizeMappings: SlotSizeMappings = {
 
 const getAdSize = (size: SizeKeys): AdSize => adSizes[size];
 
+/**
+ * Finds the ad sizes that will be used for a breakpoint given a size mapping
+ *
+ * If ad sizes are defined in the size mapping for the specified breakpoint, we use that.
+ * If no sizes are defined for the breakpoint, use the next smallest breakpoint with defined ad sizes.
+ * If no smaller breakpoints have defined ad sizes, return an empty array
+ */
+const findAppliedSizesForBreakpoint = (
```

Comment:
Could you maybe throw a couple of examples in the comment?

Reply:
Will do. There's also more examples in the tests for this function

Reply:
sweet